In [9]:
#Import Libraries
import pandas as pd
import numpy as np
import scipy.stats as stats

In [10]:
#Create DataFrame
data = [100,150,50,100,130,120,100,85,70,150,150,120,50,100,100,140,90,150,50,90,120,100,110,75,65]
df = pd.DataFrame(data, columns=["TransactionAmount"])
print(df)


    TransactionAmount
0                 100
1                 150
2                  50
3                 100
4                 130
5                 120
6                 100
7                  85
8                  70
9                 150
10                150
11                120
12                 50
13                100
14                100
15                140
16                 90
17                150
18                 50
19                 90
20                120
21                100
22                110
23                 75
24                 65


In [12]:
#Central Tendency
import statistics
print("mean : ", statistics.mean(data))
#print median
print("median : ", statistics.median(data))

mean :  102.6
median :  100


#analisis
1. rata-rata penjualan setelah training adalah 102.6 mengalami peningkatan sebesar 2.6 yang awalnya adalah 100
2. nilai mediannya adalah 100

In [13]:
#Variability
#import numpy as np
print(np.var(data))
print(np.std(data))
print(np.max(data) - np.min(data))
print(np.quantile(data, 0.25))
print(np.quantile(data,0.5))
print(np.quantile(data, 0.75))

972.2399999999999
31.18076330047101
100
85.0
100.0
120.0


analisis
1. Variance yang didapatkan adalah 972.2399999999999
2. Standard deviansi adalah 31.18076330047101
3. Range adalah 100
4. Q1 = 85 Q2 = 100 dan Q3 = 120

In [8]:
#Define H0 and H1
#H0 = rata-rata sebelum training 100
#H1 = rata-rata setelah training > 100

In [5]:
#Define alpha = 0.05

In [14]:
#Calculate t statistics and P-value
#Perform one sample t-test
stats.ttest_1samp(a=data, popmean=100)

TtestResult(statistic=0.4085001556802841, pvalue=0.6865284813438117, df=24)

In [ ]:
#statistic= 0.40 , pvalue= 0.68

In [11]:
#Conclude with alpha and p-value
#Conclusion: p-value jauh dari alpha maka gagal menolak H0
#Training tidak efektif karena rata-rata setelah training masih di nilai 100

In [12]:
#Conclude with t statistics and critical region
#Find the critical region

In [15]:
#H1: larger(right side)
stats.t.ppf(1-0.05, len(data)-1)

1.7108820799094275

In [14]:
#Critical region result = 1.7
# nilai statistic di luar critical region
#Conclusion: Training tidak efektive,rata-rata penjualan masih dinilai 100.